### DISCUSSION TITLE
視界内のモンスター一覧からアイテムに擬態するモンスターを除く

### DISCUSSION THREAD

**Discussion by habu1010** [hengband/hengband#89] [Ideas]
Discussions機能の使い勝手を見るのを兼ねて、このような案を上げてみます。

視界内のモンスター一覧に表示されるものから、アイテムに擬態するモンスターを除くという案です。

Discordで、視界内のモンスター一覧の自動更新が実装されたからデスソードで死ぬことはなくなったな、というような話がありました。
確かに、視界内のモンスター一覧の自動更新機能が取り込まれて便利になった反面、アイテムに擬態するモンスターの存在意義が薄れた感じがします。
特にデスソードやクローカーなんかは初見殺し・油断殺しというフレーバー的な存在でもあり、それがスポイルされてしまっているのは残念な感じがあります。
その時、本家（勝手版かAngbandかはわかりませんが）ではデスソード等は表示されないという話もあったような気がします。

**Comment by Hourier** [comment]
#147 の再掲：

案としては賛成で、具体的に誰を隠すかは更に議論の余地がありそうです
Wikiでいうところのミミック・クリーピングコイン・スペースモンスター辺りを表示させないようにする？
(最後は微妙な気もするが一応議題に挙げておきたい)

**Comment by Hourier** [comment]
過去には作りかけ？ のコードもあった模様：
https://github.com/hengband/hengband/issues/147#issuecomment-779233425

**Comment by daradarach** [comment]
x+++++++(略)で見れる情報を一括表示しているに過ぎないので中の人の慎重さに依存するわけだと思いますが
視界内モンスター一覧を表示しててもデスソードで死んだ@としては見れても特に問題ないのかなと。

**Reply by habu1010** [reply]
デスソードなどを表示したくないというのは、個人的には油断を狙って殺したいというよりはフレーバー的な側面が強いですかね。
ポツンと剣がおいてある部屋に入った時に、一覧に「デスソード」って1体だけ表示されるのがすごく味気ない感じがします。

**Reply by iks3** [reply]
フレーバー面であれば、装備品シンボル( ()、[]、|、/、 \、= )、箱(&)、像(`)、通貨($)、道具類( !、? )は出ない方がよさそうですね。食料系( , )と床系( . )と壁系( #、% )は個々に対応になりそうですが。

**Comment by HiddenNameZ** [comment]
ストブリのような動くミミックが表示されないのは若干違和感がありそうなので、
動かない、あるいは、眠っているミミックモンスターのみ表示しない方が良いのではないでしょうか
また、一部の石像系のgシンボル(キリス・ウンゴルの番人など)も表示しない方がフレーバー的には"らしい"かもしれません

**Comment by habu1010** [comment]
ここまでの意見を反映して、 #371 に draft PR を作成してみました。
一覧に表示しないモンスターは以下の条件に合致したものです。

- モンスターのシンボルが /|\\()[]=$,.!?&`#% のいずれかである
- 動かない。もしくは動く場合でも寝ている

gシンボルについては、変愚蛮怒においてはモンスターである事が確定しているので、対象外としています。
キリス・ウンゴルの番人など擬態しているとされるようなgは、シンボルのほうを`(像)に変えたほうがいいかもしれません。

**Reply by habu1010** [reply]
現在のところ、一般化できる条件のみに絞っており、まだ特殊な条件は入っていません。
現状の条件でこれはどうかな、と思うのは、ユニークモンスターのそれが動くため条件に合致しない事です。
このモンスターは完全に「見えない」という部分をフィーチャーしているので、一覧にも表示されないほうがいいかな？と思います。

**Reply by iks3** [reply]
『それ』とビハインダーあたりですかね。

**Comment by habu1010** [comment]
PR #371 を更新して、次のような仕様を加えました。

- 感知魔法/スキルによって感知した場合は(感知中のみ)擬態モンスターでもモンスター一覧に表示する → 積極的に探しに行っているので見つかるのが自然と考えられるため。また感知魔法/スキルは目で探しているわけではないので擬態に惑わされることがなさそう。
- ESPで感知している場合はモンスター一覧に表示する → ESPでの感知は目で見るのとは違って擬態に騙されることはないと考えられるため。

これで擬態モンスターを視界内のモンスター一覧から除く仕様としてはかなり自然になったと思いますが、どうでしょうか。

**Reply by habu1010** [reply]
最終的に固まったこの仕様であらかた賛同を得られたかと思いますので、一旦これでdevelopブランチにマージしました。
しばらく様子を見てもんだいなければ本DiscussionはLockしようと思います。

**Comment by oguchi2918** [comment]
本Discussionとはズレますが、
視界内モンスター一覧表示を(別ウィンドウではなく)メインウィンドウに重ねて表示させるようなキーコマンドは存在するのでしょうか？
少なくてもヘルプの記述には存在しない模様ですが。

**Reply by deskull-m** [reply]
あくまでサブウィンドウを想定した実装のはずです。

**Reply by oguchi2918** [reply]
個人的にサブウィンドウは使用しない派なので、
持ち物や装備のように重ねるor^MからMAP全体表示での切り替えに組み込む表示形式の実装を希望します。
＃ゲームはフルスクリーン派

まあ割と面倒or低需要かもしれないので、後回しで構いませんが。

**Reply by deskull-m** [reply]
対応するのは大分先になりそうですがIssue化はしておきます。
